frontend/shared/components/src/useReview.ts

```diff
@@ -0,0 +1,107 @@
+import { Decoder } from "@simonbackx/simple-encoding";
+import { CenteredMessage, Toast, useContext, useOrganization } from "@stamhoofd/components";
+import { useOrganizationManager, useRequestOwner } from "@stamhoofd/networking";
+import {
+    OrganizationRegistrationPeriod,
+    SetupStep,
+    SetupStepType,
+} from "@stamhoofd/structures";
+import { computed, ComputedRef, readonly, Ref, ref } from "vue";
+
+export type UseReview = {
+    readonly step: ComputedRef<SetupStep | undefined>;
+    readonly checkboxValue: Ref<boolean>;
+    readonly save: () => Promise<boolean>;
+    readonly setValue: (value: boolean) => void;
+    readonly type: SetupStepType;
+    readonly isSaving: Ref<boolean>,
+    readonly hasChanges: ComputedRef<boolean>;
+    readonly isDone: ComputedRef<boolean>;
+    readonly overrideIsDone: Ref<boolean | null>;
+    readonly forceUpdate: () => Promise<void>;
+}
+
+export function useReview(type: SetupStepType): UseReview {
+    const organizationManager = useOrganizationManager();
+    const organization$ = useOrganization();
+    const context = useContext();
+    const owner = useRequestOwner();
+    const step = computed(() => organization$.value?.period.setupSteps.get(type));
+    let originalCheckboxValue = !!step.value?.isReviewed;
+    const checkboxValue = ref(originalCheckboxValue);
+    const isSaving = ref(false);
+    const hasChanges = computed(() => originalCheckboxValue !== checkboxValue.value);
+    const overrideIsDone = ref<boolean | null>(null);
+    const isDone = computed(() => overrideIsDone.value !== null ? overrideIsDone.value : !!step.value?.isDone);
+
+    const updateReviewedAt = async (isReviewed: boolean): Promise<boolean> => {
+        if(!hasChanges.value) return true;
+
+        const periodId = organization$.value?.period.id;
+        if (!periodId) {
+            return false;
+        }
+
+        try {
+            isSaving.value = true;
+            const updatedPeriod =
+                await context.value.authenticatedServer.request({
+                    method: "POST",
+                    path: `/organization/registration-period/${periodId}/setup-steps/review`,
+                    body: {type, isReviewed},
+                    shouldRetry: false,
+                    owner,
+                    decoder:
+                        OrganizationRegistrationPeriod as Decoder<OrganizationRegistrationPeriod>,
+                });
+
+            organizationManager.value.updatePeriods([updatedPeriod.data]);
+        } catch (error) {
+            // todo: translate
+            new Toast(
+                "Het voltooien van deze stap is mislukt.",
+                "error red"
+            ).show();
```

Comment:
```suggestion
            Toast.fromError(error).show()
```
Dit zorgt ervoor dat de backend de kans krijgt om zelf een bericht te kiezen, maar ook dat bepaalde errors die uitlegbaar zijn (bv. geen internetverbinding) correct gecommuniceerd worden